src/cljs/rems/user_settings.cljs

```diff
@@ -80,10 +80,32 @@
                    (when first-time? ; to avoid an infinite loop if the settings fail to save (e.g. unsupported language in cookies)
                      [[:check-if-should-save-language!]]))})))
 
+(rf/reg-event-fx
+ :flash-message-error
+ (fn [_ user-settings-error]
+   ;; TODO: refactor this part
+   ;; this doesnt work
+   ;; (flash-message/default-error-handler :top "Fetch user settings")
+   ;; this works
+   (if (= 401 (:status user-settings-error)) 
+      (flash-message/show-default-error! :top (str "Fetch user settings" (:status user-settings-error)))
+     (flash-message/show-default-error! :top (str "Fetch user settings")))
+   {}))
+
+(rf/reg-event-fx
+ :loaded-user-settings-fail
```

Comment:
What's the difference between `-fail` and `-error`? Could they be combined?

You could also consider just inlining the event(s) into the `:error-handler`. We have something like that in `rems.application/save-attachment` for example.

Reply:
I think I just got confused with the initial 401 suggestion